core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -140,110 +141,146 @@ public TopicOverview getTopicOverview(String topicNameSearch) {
     int tenantId = commonUtilsService.getTenantId(getUserName());
-    TopicOverview topicOverview = new TopicOverview();
-    topicOverview.setTopicExists(true);
+    SchemaOverview schemaOverview = new SchemaOverview();
+    schemaOverview.setTopicExists(true);
     boolean retrieveSchemas = true;
     updateAvroSchema(
-        topicNameSearch, schemaVersionSearch, handleDb, retrieveSchemas, topicOverview, tenantId);
-    return topicOverview;
+        topicNameSearch, schemaVersionSearch, handleDb, retrieveSchemas, schemaOverview, tenantId);
+    return schemaOverview;
   }
 
   private void updateAvroSchema(
       String topicNameSearch,
       String schemaVersionSearch,
       HandleDbRequests handleDb,
       boolean retrieveSchemas,
-      TopicOverview topicOverview,
+      SchemaOverview schemaOverview,
       int tenantId) {
-    if (topicOverview.isTopicExists() && retrieveSchemas) {
+    if (schemaOverview.isTopicExists() && retrieveSchemas) {
       List<Map<String, String>> schemaDetails = new ArrayList<>();
-      Map<String, String> schemaMap = new HashMap<>();
+      schemaOverview.setSchemaDetails(schemaDetails);
+      Map<String, List<Integer>> schemaVersions = new HashMap<>();
+      schemaOverview.setAllSchemaVersions(schemaVersions);
+      schemaOverview.setLatestVersion(new HashMap<>());
       List<Env> schemaEnvs = handleDb.selectAllSchemaRegEnvs(tenantId);
       Object dynamicObj;
       Map<String, Object> hashMapSchemaObj;
       String schemaOfObj;
+
       for (Env schemaEnv : schemaEnvs) {
         try {
+          log.debug("UpdateAvroSchema - Process env {}", schemaEnv);
+          Map<String, String> schemaMap = new HashMap<>();
           KwClusters kwClusters =
               manageDatabase
                   .getClusters(KafkaClustersType.SCHEMA_REGISTRY, tenantId)
                   .get(schemaEnv.getClusterId());
+
           SortedMap<Integer, Map<String, Object>> schemaObjects =
               clusterApiService.getAvroSchema(
                   kwClusters.getBootstrapServers(),
                   kwClusters.getProtocol(),
                   kwClusters.getClusterName() + kwClusters.getClusterId(),
                   topicNameSearch,
                   tenantId);
-
-          Integer latestSchemaVersion = schemaObjects.firstKey();
-          Set<Integer> allVersions = schemaObjects.keySet();
-          List<Integer> allVersionsList = new ArrayList<>(allVersions);
-
-          try {
-            if (schemaVersionSearch != null
-                && latestSchemaVersion == Integer.parseInt(schemaVersionSearch)) {
-              schemaVersionSearch = "";
+          // If the schemaObject is null ie does not exist do not try to manipulate it.
+          if (schemaObjects != null && !schemaObjects.isEmpty()) {
+
+            Integer latestSchemaVersion = schemaObjects.firstKey();
+            schemaOverview.getLatestVersion().put(schemaEnv.getName(), latestSchemaVersion);
```

Comment:
do we retrieve schemas of all the environments or just the base environment ?

Reply:
It is better to retrieve only base, and if user would like to view all others, they can click on something which makes another api call ?